[RUM-MCP] Mark rum toolset as GA and document new tools (#37848)
DataDog/documentation: content/en/mcp_server/tools.md

Tools for [Real User Monitoring][58], including resolving applications, summarizing performance, surfacing aggregated insights for views, exploring metrics, and inspecting application configuration.

### `search_datadog_rum_events`
*Toolset: **core**, **rum***\
*Permissions Required: `RUM Apps Read`*\
Search and retrieve raw Datadog RUM events using advanced query syntax. Use for inspecting individual events or debugging specific user experience issues. For counts, aggregations, or grouped analysis, use `aggregate_rum_events`.

- Show JavaScript errors and console warnings in RUM.
- Find pages that are loading slowly (more than 3 seconds).
- Show recent user interactions on product detail pages.

### `aggregate_rum_events`
*Toolset: **core**, **rum***\
*Permissions Required: `RUM Apps Read`*\

> Aggregates Datadog RUM events to compute counts, sums, averages, min, max, cardinality, and percentiles (P50–P99), with optional grouping by fields or time intervals. Use for aggregated analysis such as session counts over time, error counts by page, or p95 loading times by browser. For raw event inspection, use `search_datadog_rum_events`.

Reviewer: ```suggestion
Aggregates Datadog RUM events to compute counts, sums, averages, min, max, cardinality, and percentiles (p50–p99), with optional grouping by fields or time intervals. Use for aggregated analysis such as session counts over time, error counts by page, or p95 loading times by browser. For raw event inspection, use `search_datadog_rum_events`.
```

For consistency (you use lowercase p elsewhere)

- List the retention filters configured on the "checkout-web" application.
- What retention filters do I have on my main RUM app?

### `upsert_rum_metric`
*Toolset: **rum***\
*Permissions Required: `RUM Generate Metrics` and `RUM Apps Read`*\

> Creates or updates a RUM custom metric. Immutable fields (event type, aggregation, path) cannot be changed after creation — delete and recreate to change them. Requires `confirm: true`.

Reviewer: ```suggestion
Creates or updates a RUM custom metric. Immutable fields (event type, aggregation, path) cannot be changed after creation. To change them, delete and recreate the metric. Requires `confirm: true`.
```

We try to avoid using em dashes like this (partly style preference, partly because it can be a hallmark of AI use)

- Create a count metric for checkout errors grouped by service: metric ID `rum.error.checkout_errors`, event type `error`, filter `@error.resource.url:*checkout*`, group by `@service`.
- Update the filter on an existing RUM metric to narrow it to production traffic only.

### `delete_rum_metric`
*Toolset: **rum***\
*Permissions Required: `RUM Generate Metrics` and `RUM Apps Read`*\

> Permanently deletes a RUM custom metric by ID. Only custom metrics can be deleted (not out-of-the-box metrics). Use `search_rum_metrics` with `type=custom` to confirm the metric ID first. Requires `confirm: true`.

Reviewer: ```suggestion
Permanently deletes a RUM custom metric by ID. Only custom metrics can be deleted (not out-of-the-box metrics). Use `search_rum_metrics` with `type=custom` to verify the metric ID before deleting. Requires `confirm: true`.
```

- Delete the RUM custom metric `rum.error.checkout_errors`.

> - Remove the metric `rum.view.lcp_by_country` after confirming with the user.

Reviewer: This phrasing seems funny to me; it doesn't seem natural for you (the user) to write the prompt this way, referring to yourself. But let me know if there's something specific you want to convey with this example.
